pywbemtools/pywbemcli/_cmd_statistics.py

```diff
@@ -0,0 +1,347 @@
+
+OBJMGR_CLN = "CIM_ObjectManager"
+OBJMGR_STAT_PROPERTY_NAME = "GatherStatisticalData"
+
+
+def test_conn_exists(context):
+    """
+    Test if server defined.
+
+    Returns if server defined
+
+    Exception:
+       click.Exception if no current server.
+    """
+    if not context.conn:
+        raise click.ClickException(
+            'No server defined in current context.')
+
+
+def get_objmgr_inst(context):
+    """
+    Return the state of statistics gathering on the server.
+    """
+    conn = context.conn
+    try:
+        interop_ns = context.wbem_server.interop_ns
+    except Error as er:
+        raise click.ClickException(
+            'Server connection failed. Exception: {}'.format(er))
+
+    try:
+        objmanager_insts = conn.EnumerateInstances(OBJMGR_CLN,
+                                                   namespace=interop_ns)
+
+        # There must be only one CIM_ObjectManager instance. None causes
+        # failure and more than 1 causes warning
+        if not objmanager_insts:
+            raise click.ClickException(
+                'No instances of class {} found on server'.format(OBJMGR_CLN))
+
+        if len(objmanager_insts) > 1:
+            warning_msg(
+                "Server returned multiple {} instances. Using first.".
+                format(objmanager_insts[0].classname))
+
+    except Error as er:
+        raise click.ClickException(
+            'No {} found on server in namespace {}. Error: {}'.
+            format(OBJMGR_CLN, interop_ns, er))
+    objmgr_inst = objmanager_insts[0]
+
+    return objmgr_inst
+
+
+def set_server_statistics(context, desired_state):
+    """
+    Set the server statistics state to on or off, enabling or disabling
+    the statistics.
+
+    Parameters:
+      desired_state (:class": `boolean)
+        Value desired for the GatherStasticalData property
+    """
+    conn = context.conn
+    interop_ns = context.wbem_server.interop_ns
+
+    objmgr = get_objmgr_inst(context)
+    cur_state = objmgr[OBJMGR_STAT_PROPERTY_NAME]
+    if cur_state == desired_state:
+        click.echo("Server already in state {}".format(desired_state))
+        return
+
+    objmgr[OBJMGR_STAT_PROPERTY_NAME] = desired_state
+
+    # Otherwise modify the property in the server
+    try:
+        conn.ModifyInstance(objmgr, PropertyList=[OBJMGR_STAT_PROPERTY_NAME],
+                            IncludeQualifiers=False)
+    except Error as ce:
+        raise click.ClickException('Modification of GatherStasticalData '
+                                   'property on server failed {}'.format(ce))
+
+    modified_objmgr = conn.GetInstance(objmgr.path)
+
+    if modified_objmgr[OBJMGR_STAT_PROPERTY_NAME] != desired_state:
+        raise click.ClickException(
+            'Modification failed classname {} on server namespace {}'.
+            format(OBJMGR_CLN, interop_ns))
+
+    click.echo("Server GatherStatisticalData set {}". format(desired_state))
```

Comment:
**DISCUSSION:** I don't like it that this message mentions the GatherStatisticalData property. The semantical level would be that the server has enabled the gathering of server statistics and the returning of server response times. On the other hand, what if a server only enables one of the two mechanisms as a result of this property? Then mentioning it may be helpful. On the other hand, we could say that this is clarified in the documentation and the message should assume that both is done.

Reply:
Terminology on the three pieces of statistics.  1. Client kept statistics. 2.  server provided but client kept (The operation time passed with responses, 3. Server kept statistics.

We need to be sure we are clear which statistics we are talking about in all cases. 

Lets change this messages to "Server statistics gathering set {} 

Change to  make the {} on/off rather true false.

Review to sort out the complexity of the http returned statistics so user understands it involves both the server and client.

Reply:
Rewrote much of overview and tried to keep to consistent terminology.  We use term "pywbemcli gatherer' or pywbemcli maintained" and "wbem_server gatherer"